org.eclipse.paho.android.service/src/main/java/org/eclipse/paho/android/service/MqttAndroidClient.java

```diff
@@ -321,7 +326,21 @@ public IMqttToken connect(MqttConnectOptions options, Object userContext, IMqttA
         if (mqttService == null) { // First time - must bind to the service
             Intent serviceStartIntent = new Intent();
             serviceStartIntent.setClassName(myContext, SERVICE_NAME);
-            Object service = myContext.startService(serviceStartIntent);
+
+            Object service = null;

```

Comment:
`service` is never assigned a value

Reply:
You are right, thank you for pointing that out. I have just pushed a commit with a fix.